core-bundle/src/Resources/public/core.js

```diff
@@ -992,10 +993,50 @@ var Backend =
 	},
 
 	/**
-	 * Get the current scroll offset and store it in a cookie
+	 * Store the current scroll offset in sessionStorage
 	 */
 	getScrollOffset: function() {
-		document.cookie = "BE_PAGE_OFFSET=" + window.getScroll().y + "; path=" + (Contao.path || '/');
+		window.sessionStorage.setItem('contao_backend_offset', parseInt(window.getScroll().y, 10) + window.location);
+	},
+
+	/**
+	 * Scrolls to the current offset if it was defined and gives the header the "down" CSS class.
+	 * It also removes the legacy BE_PAGE_OFFSET cookie.
+	 */
+	initScrollOffset: function() {
+		// Make sure we always kill the legacy cookie here
+		// That way it can be sent by the server but it wont be resent by the
+		// client in the next request
+		Cookie.dispose('BE_PAGE_OFFSET');
+
+		var item = window.sessionStorage.getItem('contao_backend_offset'), additionalOffset = 0;
+
+		if (!item) {
+			return;
+		}
+
+		// Remove session storage item again
+		window.sessionStorage.removeItem('contao_backend_offset');
+
+		var chunks = item.split(/^(\d*)/);
```

Comment:
I think it would be more efficient to use a trailing character such as `|` so we do not need the regex, wouldn't it?

Reply:
I think the regex ist pretty fast because it's super simple. Otherwise we'd need to find a character that is not allowed in an URL.

Reply:
I ran a quick test with the null byte as separator (probably not the best choice) and it was 18% faster: http://jsben.ch/hyQhS

<img width="680" alt src="https://user-images.githubusercontent.com/1192057/57124613-b4549c80-6d86-11e9-802f-80ee355796f4.png">

Reply:
Let's use this then, do you want to push yourself as it was your idea anyway? :)

Reply:
Not needed anymore, so I mark this as resolved.